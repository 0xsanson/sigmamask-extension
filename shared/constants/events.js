export const CONTROLLER_CONNECTION_EVENTS = {
  UPDATE: 'update',
  SEND_UPDATE: 'sendUpdate',
  ACTION: 'action',
  SEND_ACTION: 'sendAction',
  HARDWARE_CALL: 'hardwareCall',
  SEND_HARDWARE_CALL: 'sendHardwareCall',
};
