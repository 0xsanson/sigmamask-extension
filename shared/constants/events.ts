export enum ControllerConnectionEvents {
  update = 'update',
  sendUpdate = 'sendUpdate',
  action = 'action',
  sendAction = 'sendAction',
  hardwareCall = 'hardwareCall',
  sendHardwareCall = 'sendHardwareCall',
}
