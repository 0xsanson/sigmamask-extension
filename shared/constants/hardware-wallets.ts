/**
 * Hardware wallets supported by MetaMask.
 */
export enum HardwareKeyringType {
  ledger = 'Ledger Hardware',
  trezor = 'Trezor Hardware',
  lattice = 'Lattice Hardware',
  qr = 'QR Hardware Wallet Device',
}

export enum HardwareKeyringNames {
  ledger = 'Ledger',
  trezor = 'Trezor',
  lattice = 'Lattice1',
  qr = 'QR',
}

export enum HardwareDeviceNames {
  ledger = 'ledger',
  trezor = 'trezor',
  lattice = 'lattice',
  qr = 'QR Hardware',
}

export enum HardwareTransportStates {
  none = 'NONE',
  verified = 'VERIFIED',
  deviceOpenFailure = 'DEVICE_OPEN_FAILURE',
  unknownFailure = 'UNKNOWN_FAILURE',
}

export enum HardwareAffiliateLinks {
  ledger = 'https://shop.ledger.com/?r=17c4991a03fa',
  gridplus = 'https://gridplus.io/?afmc=7p',
  trezor = 'https://shop.trezor.io/product/trezor-one-black?offer_id=35&aff_id=11009',
  keystone = 'https://shop.keyst.one/?rfsn=6088257.656b3e9&utm_source=refersion&utm_medium=affiliate&utm_campaign=6088257.656b3e9',
  airgap = 'https://airgap.it/',
  coolwallet = 'https://www.coolwallet.io/',
  dcent = 'https://dcentwallet.com/',
}

export enum HardwareAffiliateTutorialLinks {
  ledger = 'https://support.ledger.com/hc/en-us/articles/4404366864657-Set-up-and-use-MetaMask-to-access-your-Ledger-Ethereum-ETH-account?docs=true',
  gridplus = 'https://docs.gridplus.io/setup/metamask',
  trezor = 'https://wiki.trezor.io/Apps:MetaMask',
  keystone = 'https://support.keyst.one/3rd-party-wallets/eth-and-web3-wallets-keystone/bind-metamask-with-keystone',
  airgap = 'https://support.airgap.it/guides/metamask/',
  coolwallet = 'https://www.coolwallet.io/metamask-step-by-step-guides/',
  dcent = 'https://medium.com/dcentwallet/dcent-wallet-now-supports-qr-based-protocol-to-link-with-metamask-57555f02603f',
}

/**
 * Used for setting the users preference for ledger transport type
 */
export enum LedgerTransportTypes {
  live = 'ledgerLive',
  webhid = 'webhid',
  u2f = 'u2f',
}

export enum WebHIDConnectedStatuses {
  connected = 'connected',
  notConnected = 'notConnected',
  unknown = 'unknown',
}

export const LEDGER_USB_VENDOR_ID = '0x2c97';

export const U2F_ERROR = 'U2F';

export const LEDGER_LIVE_PATH = `m/44'/60'/0'/0/0`;
export const MEW_PATH = `m/44'/60'/0'`;
export const BIP44_PATH = `m/44'/60'/0'/0`;
export const LEDGER_HD_PATHS = [
  { name: 'Ledger Live', value: LEDGER_LIVE_PATH },
  { name: 'Legacy (MEW / MyCrypto)', value: MEW_PATH },
  { name: `BIP44 Standard (e.g. MetaMask, Trezor)`, value: BIP44_PATH },
];

export const LATTICE_STANDARD_BIP44_PATH = `m/44'/60'/0'/0/x`;
export const LATTICE_LEDGER_LIVE_PATH = `m/44'/60'/x'/0/0`;
export const LATTICE_MEW_PATH = `m/44'/60'/0'/x`;
export const LATTICE_HD_PATHS = [
  {
    name: `Standard (${LATTICE_STANDARD_BIP44_PATH})`,
    value: LATTICE_STANDARD_BIP44_PATH,
  },
  {
    name: `Ledger Live (${LATTICE_LEDGER_LIVE_PATH})`,
    value: LATTICE_LEDGER_LIVE_PATH,
  },
  { name: `Ledger Legacy (${LATTICE_MEW_PATH})`, value: LATTICE_MEW_PATH },
];

export const TREZOR_TESTNET_PATH = `m/44'/1'/0'/0`;
export const TREZOR_HD_PATHS = [
  { name: `BIP44 Standard (e.g. MetaMask, Trezor)`, value: BIP44_PATH },
  { name: `Trezor Testnets`, value: TREZOR_TESTNET_PATH },
];

export const HD_PATHS = {
  ledger: LEDGER_HD_PATHS,
  lattice: LATTICE_HD_PATHS,
  trezor: TREZOR_HD_PATHS,
};

export const DEFAULT_HD_PATHS = {
  trezor: BIP44_PATH,
  ledger: LEDGER_LIVE_PATH,
  lattice: BIP44_PATH,
};
