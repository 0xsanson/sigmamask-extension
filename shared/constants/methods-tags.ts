/**
 * This is a list of methods that require the globally selected network
 * to match the dapp selected network before being processed. These can
 * be for UI/UX reasons where the currently selected network is displayed
 * in the confirmation even though it will be submitted on the correct
 * network for the dapp. It could also be that a method expects the
 * globally selected network to match some value in the request params itself.
 */
export const methodsRequiringNetworkSwitch = Object.freeze([
  'eth_sendTransaction' as const,
  'eth_sendRawTransaction' as const,
  'wallet_switchEthereumChain' as const,
  'wallet_addEthereumChain' as const,
  'wallet_watchAsset' as const,
  'eth_signTypedData_v4' as const,
  'personal_sign' as const,
]);

/**
 * This is a list of methods that can cause a confirmation to be
 * presented to the user. Note that some of these methods may
 * only sometimes cause a confirmation to appear.
 */
export const methodsWithConfirmation = Object.freeze([
  ...methodsRequiringNetworkSwitch,
  'wallet_requestPermissions' as const,
  'wallet_requestSnaps' as const,
  'wallet_snap' as const,
  'wallet_invokeSnap' as const,
  'eth_decrypt' as const,
  'eth_sign' as const,
  'eth_requestAccounts' as const,
  'eth_getEncryptionPublicKey' as const,
]);
