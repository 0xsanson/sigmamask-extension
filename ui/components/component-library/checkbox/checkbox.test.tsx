import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { FontWeight } from '../../../helpers/constants/design-system';
import { Checkbox } from '.';

describe('Checkbox', () => {
  it('should render the Checkbox without crashing', () => {
    const { getByRole, container } = render(<Checkbox />);
    expect(getByRole('checkbox')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render isChecked', () => {
    const { getByRole, getByTestId } = render(
      <Checkbox isChecked={true} iconProps={{ 'data-testid': 'check-bold' }} />,
    );
    expect(getByRole('checkbox')).toBeChecked();
    expect(window.getComputedStyle(getByTestId('check-bold')).maskImage).toBe(
      `url('./images/icons/check-bold.svg')`,
    );
  });

  it('should render isIndeterminate', () => {
    const { getByRole, getByTestId } = render(
      <Checkbox
        isIndeterminate={true}
        iconProps={{ 'data-testid': 'minus-bold' }}
      />,
    );
    expect(getByRole('checkbox').getAttribute('data-indeterminate')).toBe(
      'true',
    );
    expect(window.getComputedStyle(getByTestId('minus-bold')).maskImage).toBe(
      `url('./images/icons/minus-bold.svg')`,
    );
  });

  it('should render checkbox with label', () => {
    const { getByText } = render(<Checkbox label="Option 1" />);
    expect(getByText('Option 1')).toBeDefined();
  });

  it('should render checkbox with label that has additional textProps for bold text', () => {
    const { getByText } = render(
      <Checkbox label="Option 1" textProps={{ fontWeight: FontWeight.Bold }} />,
    );
    expect(getByText('Option 1')).toHaveClass('mm-text--font-weight-bold');
  });

  it('should render checkbox with id and label has matching htmlFor', () => {
    const { getByText, getByRole } = render(
      <Checkbox label="Option 1" id="option-1" />,
    );
    const checkbox = getByRole('checkbox');

    expect(checkbox).toHaveAttribute('id', 'option-1');
    expect(getByText('Option 1')).toHaveAttribute('for', 'option-1');
  });

  test('Checkbox component is disabled when isDisabled is true', () => {
    const { getByLabelText } = render(
      <Checkbox label="Option 1" id="option-1" isDisabled={true} />,
    );

    const checkbox = getByLabelText('Option 1');

    expect(checkbox).toBeDisabled();
    expect(checkbox).toHaveClass('mm-checkbox--disabled');
  });

  test('Checkbox component is readOnly when isReadOnly is true', () => {
    const { getByLabelText } = render(
      <Checkbox label="Option 1" id="option-1" isReadOnly={true} />,
    );

    const checkbox = getByLabelText('Option 1');

    expect(checkbox).toHaveAttribute('readonly');
    expect(checkbox).toHaveClass('mm-checkbox--readonly');
  });

  it('Checkbox component fires onChange function when clicked', () => {
    const onChange = jest.fn();

    const { getByTestId } = render(
      <Checkbox data-testid="checkbox" onChange={onChange} />,
    );

    const checkbox = getByTestId('checkbox');

    fireEvent.click(checkbox);

    expect(onChange).toHaveBeenCalled();
  });

  test('Checkbox component is required when isRequired is true', () => {
    const { getByLabelText } = render(
      <Checkbox label="Option 1" id="option-1" isRequired={true} />,
    );

    const checkbox = getByLabelText('Option 1');

    expect(checkbox).toHaveAttribute('required');
  });

  test('Checkbox component renders with the correct title attribute', () => {
    const { getByLabelText } = render(
      <Checkbox label="Option 1" id="option-1" title="pineapple" />,
    );

    const checkbox = getByLabelText('Option 1');

    expect(checkbox).toHaveAttribute('title', 'pineapple');
  });

  test('Checkbox component renders with the correct title attribute used from the label', () => {
    const { getByLabelText } = render(
      <Checkbox label="Option 1" id="option-1" />,
    );

    const checkbox = getByLabelText('Option 1');

    expect(checkbox).toHaveAttribute('title', 'Option 1');
  });

  test('Checkbox component renders with the correct title attribute used from the id', () => {
    const { getByRole } = render(<Checkbox id="option-1" />);

    const checkbox = getByRole('checkbox');

    expect(checkbox).toHaveAttribute('title', 'option-1');
  });

  test('Checkbox component renders with the correct name attribute', () => {
    const { getByRole } = render(<Checkbox name="option-1" />);

    const checkbox = getByRole('checkbox');

    expect(checkbox).toHaveAttribute('name', 'option-1');
  });
});
