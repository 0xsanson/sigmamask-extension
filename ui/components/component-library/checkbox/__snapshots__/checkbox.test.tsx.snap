// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render the Checkbox without crashing 1`] = `
<div>
  <div
    class="mm-box mm-checkbox__wrapper"
  >
    <input
      class="mm-box mm-checkbox mm-box--margin-0 mm-box--background-color-transparent mm-box--rounded-sm mm-box--border-color-border-default mm-box--border-width-2 box--border-style-solid"
      type="checkbox"
    />
  </div>
</div>
`;
